gpkit/constraints/set.py

```diff
@@ -52,28 +52,40 @@ def __getitem__(self, key):
         if isinstance(key, int):
             return list.__getitem__(self, key)
         else:
-            variables = self.variables_byname(key)
-            if not variables:
-                raise KeyError(key)
-            if variables[0].key.veckey:
-                # maybe it's all one vector variable!
-                from ..nomials import NomialArray
-                vk = variables[0].key.veckey
-                arr = NomialArray(np.full(vk.shape, np.nan, dtype="object"))
-                arr.key = vk
-                for variable in variables:
-                    if variable.key.veckey == vk:
-                        arr[variable.key.idx] = variable
-                    else:
-                        arr = None
-                        break
-                if arr is not None:
-                    return arr
-            elif len(variables) == 1:
-                return variables[0]
-            raise ValueError("multiple variables are called '%s'; use"
-                             " variables_byname('%s') to see all of them"
-                             % (key, key))
+            return self._choosevar(key, self.variables_byname(key))
+
+    def _choosevar(self, key, variables):
+        if not variables:
+            raise KeyError(key)
+        if variables[0].key.veckey:
+            # maybe it's all one vector variable!
+            from ..nomials import NomialArray
+            vk = variables[0].key.veckey
+            arr = NomialArray(np.full(vk.shape, np.nan, dtype="object"))
+            arr.key = vk
+            for variable in variables:
+                if variable.key.veckey == vk:
+                    arr[variable.key.idx] = variable
+                else:
+                    arr = None
+                    break
+            if arr is not None:
+                return arr
+        elif len(variables) == 1:
+            return variables[0]
+        raise ValueError("multiple variables are called '%s'; use"
+                         " variables_byname('%s') to see all of them"
+                         % (key, key))
+
+    def topvar(self, key):
+        "If a variable by a given name exists in the top model, return it"
+        # TODO: could this be done with unique_varkeys?
+        if not self.naming:  # pylint: disable=no-member
```

Comment:
We could give every constraintset a naming, yeah! Right now Models law claim to all variables created in them that aren't in a named submodel, whether those are created in lower-level constraintsets or not, and we could extend that by giving that naming to those constraintsets.